solarkraft/src/fetcher/storage.ts

```diff
@@ -119,6 +175,18 @@ function getEntryFilename(root: string, entry: ContractCallEntry) {
     return join(dir, `entry-${entry.txHash}.json`)
 }
 
+/**
+ * Get the filename for a contract call entry. Create the parent directory, if required.
+ *
+ * @param root storage root
+ * @param entry call entry
+ * @returns the filename
+ */
+function getFetcherStateFilename(root: string) {
+    mkdirSync(root, { recursive: true })
```

Comment:
Not really obvious that this has side-effects from the function name.

Can we just call `mkdirSync` in `saveFetcherState` instead of here?